x/authority/client/cli/tx_test.go

```diff
@@ -0,0 +1,213 @@
+
+	// set the multisig acount in the state
+	sendTokens := sdk.NewInt64Coin(s.cfg.BondDenom, 10)
+	_, err = bankcli.MsgSendExec(
+		val1.ClientCtx,
+		val1.Address,
+		authMultiSigAcc.GetAddress(),
+		sdk.NewCoins(sendTokens),
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+		fmt.Sprintf("--gas=%d", flags.DefaultGasLimit),
+	)
+
+	// Generate the unsigned multisig transaction json with the existing authority key.
+	args := []string{
+		authMultiSigAcc.GetAddress().String(),
+		newAuthMultiSigAcc.GetAddress().String(),
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+		fmt.Sprintf("--%s=true", flags.FlagGenerateOnly),
+	}
+
+	multiGeneratedTx, err := sdkcli.ExecTestCLICmd(val1.ClientCtx, cli.GetCmdReplaceAuthority(),args)
+	s.Require().NoError(err)
+
+	// Save tx to file
+	multiGeneratedTxFile := testutil.WriteToNewTempFile(s.T(), multiGeneratedTx.String())
+
+	// Sign with account1
+	val1.ClientCtx.HomeDir = strings.Replace(val1.ClientCtx.HomeDir, "simd", "simcli", 1)
+	account1Signature, err := authtest.TxSignExec(
+		val1.ClientCtx, account1.GetAddress(), multiGeneratedTxFile.Name(),
+		"--multisig", authMultiSigAcc.GetAddress().String(),
+	)
+	s.Require().NoError(err)
+
+	sign1File := testutil.WriteToNewTempFile(s.T(), account1Signature.String())
+
+	// Sign with account2
+	account2Signature, err := authtest.TxSignExec(
+		val1.ClientCtx, account2.GetAddress(), multiGeneratedTxFile.Name(),
+		"--multisig", authMultiSigAcc.GetAddress().String(),
+	)
+	s.Require().NoError(err)
+
+	sign2File := testutil.WriteToNewTempFile(s.T(), account2Signature.String())
+
+	// Does not work in offline mode.
+	_, err = authtest.TxMultiSignExec(
+		val1.ClientCtx, authMultiSigAcc.GetName(), multiGeneratedTxFile.Name(),
+		"--offline", sign1File.Name(), sign2File.Name(),
+	)
+	s.Require().EqualError(
+		err,
+		"couldn't verify signature: unable to verify single signer signature",
+	)
+
+	val1.ClientCtx.Offline = false
+	multiSigWith2Signatures, err := authtest.TxMultiSignExec(
+		val1.ClientCtx, authMultiSigAcc.GetName(), multiGeneratedTxFile.Name(),
+		sign1File.Name(), sign2File.Name(),
+	)
+	s.Require().NoError(err)
+
+	// Write the output to disk
+	signedTxFile := testutil.WriteToNewTempFile(s.T(), multiSigWith2Signatures.String())
+
+	_, err = authtest.TxValidateSignaturesExec(val1.ClientCtx, signedTxFile.Name())
+	s.Require().NoError(err)
+
+	val1.ClientCtx.BroadcastMode = flags.BroadcastBlock
+	_, err = authtest.TxBroadcastExec(val1.ClientCtx, signedTxFile.Name())
+	s.Require().NoError(err)
+
+	s.Require().NoError(s.network.WaitForNextBlock())
```

Comment:
Nice test format! Looks like something that should be adopted more widely across our codebase.

Does this test confirm that the new authority is in effect, e.g. by querying the current authority?

Reply:
> Nice test format! Looks like something that should be adopted more widely across our codebase.
That's the motive :) As committed now, this is a multisig **unit test** for the new cobra CLI command with source-level integrated debugging support. However, it could be migrated to our Emd configuration for an authentic e-Money in-process 1+ nodes integration test.


> Does this test confirm that the new authority is in effect, e.g. by querying the current authority?
No, because this is employing the default SDK in-process Tendermint node, there is no handler for the message, it acts like a mocking node for the message biz logic.